eslint-bridge/src/linting/eslint/rules/decorators/jsx-key-decorator.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube JavaScript Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { TSESTree } from '@typescript-eslint/types';
+import { Rule } from 'eslint';
+import { interceptReportForReact } from './helpers';
+
+export function decorateJsxKey(rule: Rule.RuleModule): Rule.RuleModule {
+  return interceptReportForReact(rule, (context, reportDescriptor) => {
+    // check if node has attribute containing spread operator
+    if ('node' in reportDescriptor) {
+      const { node, ...rest } = reportDescriptor;
+      if (hasSpreadOperator(node as TSESTree.Node)) {
+        return;
+      }
+      context.report({
+        node,
+        ...rest,
+      });
+    }
+  });
```

Comment:
I see there's a convention for decorators adding an exception where they define a `reportExempting()` function. It would be nice to just isolate the same logic here in a separate function. 
```suggestion
  return interceptReportForReact(rule, reportExempting(hasSpreadOperator));
}

function reportExempting(exemptionCondition: (property: TSESTree.Node) => boolean) {
  return (context: Rule.RuleContext, reportDescriptor: Rule.ReportDescriptor) => {
    // check if node has attribute containing spread operator
    if ('node' in reportDescriptor) {
      const { node, ...rest } = reportDescriptor;
      if (exemptionCondition(node as TSESTree.Node)) {
        return;
      }
      context.report({
        node,
        ...rest,
      });
    }
  };
```

Reply:
done